OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -129,7 +147,8 @@ void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newO
 			if (!Info.Sticky)
 			{
 				Faction = self.Owner.Faction.InternalName;
-				Enabled = !Info.Factions.Any() || Info.Factions.Contains(Faction);
+				UsableByFaction = !Info.Factions.Any() || Info.Factions.Contains(Faction);
+				ProductionAvailable &= UsableByFaction;
```

Comment:
Actually, on second look, this (and the version before) has a bigger problem: if `ProductionAvailable` was previously false because of `UsableByFaction` then it will never be reenabled.